managed-ledger/src/main/java/org/apache/bookkeeper/mledger/util/RangeCache.java

```diff
@@ -106,7 +106,39 @@ K getKey() {
             return localKey;
         }
 
+        /**
+         * Get the value associated with the key. Returns null if the key does not match the key.
+         *
+         * @param key the key to match
+         * @return the value associated with the key, or null if the value has already been recycled or the key does not
+         * match
+         */
         V getValue(K key) {
+            return getValueInternal(key, false);
```

Comment:
don't we always need to get value with requireSameKeyInstance=true?
In what case, are we ok with requireSameKeyInstance=false here?

Reply:
No. Essentially this single change is addressing the bug fix. 
Getting by key should also match by `.equals` check.